### DISCUSSION TITLE
SimpleTheme / FluentTheme: Migration from 0.10.18 to 11.0.0-preview

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#9314] [General]
Because I was not able to get this right, as quickly as I hoped for, I just wanted so summarize here for everybody how to migrate a freshly created (or an existing) project with resprect to themes.

If you use the VS 2022 template for project creation this will target version 0.10.18 (or similar). 
- If you want to target version 11.0.0-previewX, you just change the version in the project file
- In addition, remove the elements about ```TrimmableAssembly``` and similar from the project file
- Because the themes are now located outside of Avalonia base package, you muss add the nuget package for your theme
- I now made the experience that the Xaml loader would not work correctly anymore and show some errors not understanding the ```<FluentTheme Mode="Dark />``` element in App.axaml anymore. 
- I tried ```<StyleInclude ...>``` which made it work but should not be necessary
- Hence, one important step after the change from 0.10.18 to 11.0.0-previewX is to delete the Visual Studio cache: delete bin and obj folder (close VS before). After you restart Visual Studio, rebuild the project, everything should be fine now. 

See below the working project file and App.axaml for 11.0.0-preview3:

Project file
```
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <OutputType>WinExe</OutputType>
        <TargetFramework>net6.0</TargetFramework>
        <Nullable>enable</Nullable>
        <BuiltInComInteropSupport>true</BuiltInComInteropSupport>
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="Avalonia" Version="11.0.0-preview3" />
        <PackageReference Include="Avalonia.Desktop" Version="11.0.0-preview3" />
        <PackageReference Include="Avalonia.Themes.Simple" Version="11.0.0-preview3" />
        <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.0-preview3" />
        <!--Condition below is needed to remove Avalonia.Diagnostics package from build output in Release configuration.-->
        <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="11.0.0-preview3" />
        <PackageReference Include="XamlNameReferenceGenerator" Version="1.3.4" />
    </ItemGroup>
</Project>
```

App.axaml (with no style include, just left the comment there to show that this is not necessary)
```
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="AvaloniaApplication1.App">
    <Application.Styles>
        <!--<StyleInclude Source="avares://Avalonia.Themes.Fluent/FluentDark.xaml" />-->
        <FluentTheme Mode="Dark"/>
        <!--<SimpleTheme Mode="Dark" />-->
    </Application.Styles>
</Application>
```

Most likely most of you have figured this out already. It did take me some time and some back and forth on Avalonia Gitter (thanks there) which is the reason to again bring this up here.

**Comment by maxkatz6** [comment]
Thanks for sharing! 
There are quite a lot of changes between these two major versions, and it seems Visual Studio/.NET SDK doesn't work well with cache handling here.
